catalog/resource_external_location.go

```diff
@@ -0,0 +1,84 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type ExternalLocationsAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+func NewExternalLocationsAPI(ctx context.Context, m interface{}) ExternalLocationsAPI {
+	return ExternalLocationsAPI{m.(*common.DatabricksClient), ctx}
+}
+
+type ExternalLocationInfo struct {
+	Name           string `json:"name" tf:"force_new"`
+	URL            string `json:"url"`
+	CredentialName string `json:"credential_name"`
+	Comment        string `json:"comment,omitempty"`
+	Owner          string `json:"owner,omitempty" tf:"computed"`
+	MetastoreID    string `json:"metastore_id,omitempty" tf:"computed"`
+}
+
+func (a ExternalLocationsAPI) create(el *ExternalLocationInfo) error {
+	return a.client.Post(a.context, "/unity-catalog/external-locations", el, &el)
+}
+
+func (a ExternalLocationsAPI) get(id string) (el ExternalLocationInfo, err error) {
+	err = a.client.Get(a.context, "/unity-catalog/external-locations/"+id, nil, &el)
+	return
+}
+
+func (a ExternalLocationsAPI) update(name string, el ExternalLocationInfo) error {
+	return a.client.Patch(a.context, "/unity-catalog/external-locations/"+name, el)
```

Comment:
URLencode name? and is it really name in the API? :confused:

Reply:
yes, it's name. alphanumeric validation may be a better option than urlencoding. i don't think we allow emojis in names here...

Reply:
External locations support any SQL identifier and any character is allowed if you put them into back quotes ([doc](https://docs.databricks.com/sql/language-manual/sql-ref-identifiers.html#parameters))